File: mobile/test/location_event_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:mobile/models/location_event.dart';

void main() {
  test('LocationEvent converts to and from map', () {
    final event = LocationEvent(
      eventId: 'event-001',
      userId: 'user-001',
      deviceId: 'device-001',
      latitude: 28.6139,
      longitude: 77.2090,
      timestamp: DateTime.fromMillisecondsSinceEpoch(1000),
      sequenceNumber: 1,
    );

    final map = event.toMap();
    final restored = LocationEvent.fromMap(map);

    expect(restored.eventId, 'event-001');
    expect(restored.userId, 'user-001');
    expect(restored.deviceId, 'device-001');
    expect(restored.latitude, 28.6139);
    expect(restored.longitude, 77.2090);
    expect(restored.sequenceNumber, 1);
    expect(restored.synced, false);
  });
}

File: backend/tests/location_event_test.go
package tests

import (
	"testing"
	"time"

	"crowd-aware-navigation/backend/models"
)

func TestLocationEventCreation(t *testing.T) {
	event := models.LocationEvent{
		EventID:        "event-001",
		UserID:         "user-001",
		DeviceID:       "device-001",
		Latitude:       28.6139,
		Longitude:      77.2090,
		Timestamp:      time.Unix(1000, 0),
		SequenceNumber: 1,
	}

	if event.EventID == "" {
		t.Fatal("event ID should not be empty")
	}

	if event.UserID == "" {
		t.Fatal("user ID should not be empty")
	}

	if event.DeviceID == "" {
		t.Fatal("device ID should not be empty")
	}

	if event.Latitude == 0 || event.Longitude == 0 {
		t.Fatal("location should be populated")
	}

	if event.SequenceNumber != 1 {
		t.Fatal("sequence number should be 1")
	}
}